src/main/java/frc/robot/utils/autoaim/InterpolatingShotTree.java

```diff
@@ -51,7 +51,8 @@ public void clear() {
   private ShotData interpolate(ShotData startValue, ShotData endValue, double t) {
     return new ShotData(
         ((endValue.getAngle() - startValue.getAngle()) * t) + startValue.getAngle(),
-        ((endValue.getRPM() - startValue.getRPM()) * t) + startValue.getRPM());
+        ((endValue.getRPM() - startValue.getRPM()) * t) + startValue.getRPM(),
+        ((endValue.getFlightTime() - startValue.getFlightTime()) * t) + startValue.getFlightTime());
```

Comment:
abstract logic to somewhere else? being called in multiple places like `ShotData.java`

Reply:
removed method Interpolate from shotdata

Reply:
removing interpolate from ShotData doesnt solve this if im understanding jacobs request correctly. What you want is a method to interpolate doubles, which wpilib might have, and use that instead of a manual interpolate for each value. ShotData interpolate can live in either class although i prefer if its in ShotData